wiring/src/spark_wiring_ble.cpp

```diff
@@ -577,81 +576,81 @@ BleAdvertisingData::BleAdvertisingData(const iBeacon& beacon)
 
 size_t BleAdvertisingData::set(const uint8_t* buf, size_t len) {
     if (buf == nullptr || len == 0) {
-        selfLen_ = 0;
-        return selfLen_;
+        selfData_.clear();
+        return selfData_.size();
     }
-    len = std::min(len, (size_t)BLE_MAX_ADV_DATA_LEN);
-    memcpy(selfData_, buf, len);
-    selfLen_ = len;
-    return selfLen_;
+    selfData_.clear();
+    len = std::min(len, (size_t)BLE_MAX_ADV_DATA_LEN_EXT);
```

Comment:
We should use `BLE_MAX_ADV_DATA_LEN_EXT_CONNECTABLE` here.

Reply:
See above (non-connectable application).

Reply:
Yeah. you're right 👍 . Let's use `BLE_MAX_ADV_DATA_LEN_EXT` as the maximum length of adv data buffers.